compiler/src/dotty/tools/dotc/typer/TyperPhase.scala

```diff
@@ -58,6 +58,9 @@ class TyperPhase(addRootImports: Boolean = true) extends Phase {
   protected def discardAfterTyper(unit: CompilationUnit)(using Context): Boolean =
     unit.isJava || unit.suspended
 
+  /** Keep synchronised with `monitor` subcalls */
+  override def subPhases: List[String] = List("indexing", "typechecking", "checking java")
```

Comment:
If you want that "keep synchronized" comment to be actually upheld, you should also put a similar comments at all the places that actually do call `monitor`. Otherwise someone changing the call sites will never see this comment and hence will never update `subPhases`.

Reply:
I made the strings into context parameters, and refactored the way we tick the subphase so its actually more declarative